### DISCUSSION TITLE
SSR with jsPsych

### DISCUSSION THREAD

**Discussion by jkhartshorne** [jspsych/jsPsych#3053] [Ideas]
I'm in the midst of migrating Pushkin away from Create React App, since that's now deprecated. The main choices these days seem to be Vite or NextJS. The point of the latter is to do server-side rendering (SSR) which ... I think we don't actually want, right? That is, I assume we actually want jsPsych (and especially plugins like WebGazer) to be running client-side, since they are dynamic.

But I also may not have thought this through. Curious what you all are thinking.

Looks like @f-starace discussed using jsPsych with NextJS earlier, though it's not clear to me what the goal was there or whether it ultimately worked: https://github.com/jspsych/jsPsych/discussions/2449

**Comment by f-starace** [comment]
Hi @jkhartshorne 
Yep, that was exactly the goal. If you wrap the component that you want to use that requires a library which needs access to the DOM during initialization time (such as that version of jsPsych) you just need to load that component dynamically.

**Reply by jkhartshorne** [reply]
How do you like NextJS? Why NextJS as opposed to, say, Vite or Remix?

**Reply by f-starace** [reply]
Have no experience on Remix, I can vouch for Vite thought. It depends a lot by the scope of your project. 

For larger projects the industry seems to be moving to NextJS as the enterprise solution for react. 

If you just need a single page application and you do not need to worry about routing and server side functionality (such as multiple ways of doing data fetching, SSR or server api calls) you can also just opt for Vite.

Both are great
